### ISSUE TITLE
Cloning Claro theme doesn't create a theme that looks like Claro

### ISSUE THREAD

**Issue by jhpedemonte** [maqetta/maqetta#1465]
Cloned the **Claro** theme using the New Theme dialog, naming it "claro2".  Theme editor then opened, but the theme looked black & white.  It was not a clone of Claro.

**Comment by peller** [comment]
works for me

**Comment by billreed63** [comment]
I have not been able to reproduce this on FF or Chrome windows.

If you can still reproduce can you provide more detail as to the environment and steps

**Comment by jhpedemonte** [comment]
OK, I was able to reproduce in FF:
1. Clone theme "claro" to "claro2".
2. Clone theme editor.
3. Clone theme "claro" to "claro3".

"claro2" looks like "claro", but "claro3" comes up as black and white.

**Comment by billreed63** [comment]
I still can't reproduce this on FF lab9. Is this on your local machine? Can you zip the workspace so I could load it on my machine and see...

**Comment by jhpedemonte** [comment]
Not local, lab9 using FF.

**Comment by billreed63** [comment]
I was able to reproduce something like this, 
What I am seeing s the claro2.css file that is in cache does not have the correct contents (all the .claro2 class names are still .claro) the claro2.css file in the users work space is correct (class names are .claro2)  once I clear the browsers cache, I then started getting the correct file contents. I am not sure how the server handles the copy and serving of the cloned css file but it seems like we might have a timing issue on the server. may @childsb has some idea's

**Comment by waynevicknair** [comment]
just recreated this one on the latest external build from 'macwayne2' server running both FF and Safari on Mac. (Cloning claro theme still hangs on Mac/Chrome)

**Comment by billreed63** [comment]
I am not seeing this today, I a fairly sure this is a network issue... I will keep testing but I do not think this is a stop ship problem for M4.

**Comment by billreed63** [comment]
We need to collect information on when and how is bug is happening:

1) What browsers are you running?
2) What OS?
3) Running Maqetta local, network, single or multi user?
4) What tabs do you have open
5) What steps did you take to create the error
6) In the network panel of your browser find the theme.css file, for example if you called your theme claro2 look for claro2.css Open  claro2.css to inspect the Response contents. Copy and paste theme in a file and send them to me.  Inspect the rule to ensure that .claro has been change to your new theme name.. .claro2 If the contents are correct delete your cache and refresh to browser. Does this fix the problem?
7) Zip up the contents of your theme folder and send them to me
8) Copy and paste the console to a file and send theme to me also.

**Comment by billreed63** [comment]
I just reproduced the problem...

Here is what I am seeing...

The css file in the users workspace has been changed and is correct... But what is in the browsers cache is the old content... 
For example claro2.css has .claro in the browsers cache but .claro2 on disk
 In the net panel I have to click on claro2.css and clear browser cache to then force the browser to get the correct file....

**Comment by billreed63** [comment]
Looking at the code here is what I believe is happening:

davinci.theme.CloneTheme copys the theme files to be cloned to the new folder
Then starts modifying the names of the theme files as well as the contents of the files for the new theme name. once that is complete we then either call save() or setContent() on the files, save() calls setContent(). 

davinci.model.Resource.File.prototype.setContents() uses dojo.xhrPut to send the updated file contents to the server, dojo.xhrPut is async. So what is happening is while the server is bogged down servicing all the puts, we run off and open the theme editor before the server has processed all the theme files. So what the theme editor retrieves from the server is the old *.css files. 

What we need to do is change davinci.theme.CloneTheme wait until all the files are successfully saved on the server before opening the theme editor. We can either user dojo.deferred or add dojo.publish in a load argument to dojo.xhrPut

http://dojotoolkit.org/documentation/tutorials/1.6/deferreds/
